packages/app-store/office365calendar/lib/CalendarService.ts

```diff
@@ -159,64 +161,107 @@ export default class Office365CalendarService implements Calendar {
     }
   }
 
-  async getAvailability(
+  async getCalIds(selectedCalendars: IntegrationCalendar[]): Promise<string[]> {
+    const selectedCalendarIds = selectedCalendars
+      .filter((e) => e.integration === this.integrationName)
+      .map((e) => e.externalId)
+      .filter(Boolean);
+    if (selectedCalendarIds.length === 0 && selectedCalendars.length > 0) {
+      // Only calendars of other integrations selected
+      return Promise.resolve([]);
+    }
+
+    const ids = await (selectedCalendarIds.length === 0
+      ? this.listCalendars().then((cals) => cals.map((e_2) => e_2.externalId).filter(Boolean) || [])
+      : Promise.resolve(selectedCalendarIds));
+    return ids;
+  }
```

Comment:
this function is really not clear, could use a full rewrite

Reply:
This function just returns the calendar IDs of all the selected calendars.